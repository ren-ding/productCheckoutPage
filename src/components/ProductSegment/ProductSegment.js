import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class ProductSegment extends Component {
    render() {
        return (
            <div>
                <span>Id: {this.props.product.productId}  </span>
                <span>Name: {this.props.product.productName}  </span>
                <span>Price: {this.props.product.price}  </span>
                <button className='add-product-to-checkout' onClick={()=> this.props.addToCheckout(this.props.product.productId)}>+</button>
            </div>
        );
    }
}

ProductSegment.propTypes = {
    product: PropTypes.object.isRequired,
    addToCheckout: PropTypes.func.isRequired
};