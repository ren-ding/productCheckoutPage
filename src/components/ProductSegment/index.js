import ProductSegment from './ProductSegment';
export default ProductSegment;