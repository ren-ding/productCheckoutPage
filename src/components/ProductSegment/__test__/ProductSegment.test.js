import React from 'react';
import ProductSegment from '../ProductSegment';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });

describe('ProductSegment', ()=> {
    let props;
    let productSegment;

    beforeEach(()=> {
        props = {
            product: {},
            addToCheckout: ()=>{}
        }

        productSegment = () => shallow(<ProductSegment {...props} />);    
    });

    describe('render', () => {
       it('should render to match snapshot', () => {
           const tree = renderer.create(<ProductSegment {...props}/>).toJSON();
           expect(tree).toMatchSnapshot();
       }); 
    });

    describe('click add button', ()=> {
        it('should call addToCheckout once with productid', ()=> {
            const mockAddToCheckout = jest.fn();
            props = {
                product:{
                    productId: 'wf',
                    productName:'Workflow',
                    price:199.99
                },
                addToCheckout: mockAddToCheckout
            }
            const segment = productSegment();
            
            segment.find('.add-product-to-checkout').simulate('click');
            expect(mockAddToCheckout).toHaveBeenCalled();
            expect(mockAddToCheckout.mock.calls.length).toBe(1);
            //first call of the first argument is the productId
            expect(mockAddToCheckout.mock.calls[0][0]).toBe('wf');
        });
    });
});