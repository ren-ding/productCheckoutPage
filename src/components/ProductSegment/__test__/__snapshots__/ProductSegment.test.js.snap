// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSegment render should render to match snapshot 1`] = `
<div>
  <span>
    Id: 
      
  </span>
  <span>
    Name: 
      
  </span>
  <span>
    Price: 
      
  </span>
  <button
    className="add-product-to-checkout"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
